163 reusable platforms instruments

BryceMarshall opened the pull request:
Added persistent storage for instruments per user 
Added navbar tab to manage user instruments
updated UI on Instrument/Platform tab to reflect changes

fostermh commented:
- when trying to create a platform in the platform menu from the left I can not enter a value into the platform type or platform id fields.
- TODO: now that there can be multiple platforms we need to link instruments to platforms

BryceMarshall commented:
Merge conflicts with translation code resolved & basic functionality for platform and instruments verified.

## Bugs
- Duplicating instruments holds reference to
original
- Platform fields are not activating Required Mark
properly

fostermh commented:
> * Duplicating instruments holds reference to original - saving either entry will modify the same underlying Firebase record

I suspect a 'deepCopy' would help you out here.

BryceMarshall commented:
## Bug: Duplicate button
- Copy-paste errors:
	- Code existed to append a "(Copy)" string to make each entry unique, but it referenced fields from the Contact rather than the Instrument.
	- Updating the fields didn't resolve the issue.
### To Reproduce
- Add a saved instrument
- Duplicate it
- Save the new entry
- **Expected**
	- Two entries in saved dropdown
- **Actual**
	- New entry replaced the old one
### Resolution:
- Save Update Instrument method was referencing an identifier that had been renamed.
## Takeaways
- The 'Left List' pattern is a good candidate for code refactor- lots of duplication present.

BryceMarshall commented:
## Bug: Platform Validator 
- Updated the Required Mark on the platform page to check the presence of each field rather than using the overall platform validator
- Updated the overall platform Validator to treat platforms as a list

fostermh commented on src/components/FormComponents/Instruments.jsx:
```diff
@@ -54,55 +39,24 @@ const Instruments = ({
+    </Grid>
       <Grid item xs>
         <Grid container direction="column">
           {instrument && (
```
is this code duplicated in instrumentEditor.js?

fostermh commented on src/components/FormComponents/Platform.jsx:
```diff
@@ -1,122 +1,159 @@
-      </Grid>
+      <Grid item xs>
+        <Grid container direction="column">
+        { platform && (
```
is this code duplicated in PlatformEditor.jsx?

fostermh commented:
when selecting a platform that an instrument is part of. the label shows up as 'New platform' even though it has a name in the platform list.

<img width="738" alt="Screenshot 2024-04-17 at 10 45 47 AM" src="https://github.com/cioos-siooc/metadata-entry-form/assets/6304992/adafee6e-bbb6-4d90-b6a0-9b8b5c8ebe55">
